plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -328,15 +339,19 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
         }
       }
     }
-
-    // There was not a sampled trace header from Lambda so try from HTTP headers.
-    const httpContext = propagation.extract(
-      otelContext.active(),
-      httpHeaders,
-      headerGetter
+    const extractedContext = safeExecuteInTheMiddle(
+      () => eventContextExtractor(event),
+      e => {
+        if (e)
+          diag.error(
+            'aws-lambda instrumentation: eventContextExtractor error',
+            e
+          );
+      },
+      true
     );
-    if (trace.getSpan(httpContext)?.spanContext()) {
-      return httpContext;
+    if (trace.getSpan(extractedContext)?.spanContext()) {
+      return extractedContext;
```

Comment:
Nitpick (that I should have caught in original): Should we use `isSpanContextValid` here instead of a truthiness check?

Reply:
`isSpanContextValid` receives `SpanContext` and not the used `Context` object... 
I'd rather leave it this way right now, just as it used to be.

Reply:
I see, apologies I'm no API expert :) happy to to leave it as is

Reply:
sure, thanks for the comment!